test/options_sanity/array_uniqueness.t

```diff
@@ -0,0 +1,15 @@
+  $ source "$TESTDIR/../helper.sh"
+
+Check that relevant arrays are made unique if they contain elements, otherwise they should be left as empty arrays
```

Comment:
I think it would be clearer to organize these tests such that each test case is only testing one thing, and that we do a suite by function. As they are, the unique assertions and parsing assertions are a bit mixed.

**test/parse_options.t**

```
Reads multiple --pacman-cache options into PACMAN_CACHE

  $ parse_options --pacman-cache /home/ --pacman-cache /away/ --pacman-cache /home/ foo 2>/dev/null
  > echo "${PACMAN_CACHE[@]}"
  /home/ /away/ /home/

Reads arguments into terms

  $ parse_options foo bar foo 2>/dev/null
  > echo "${terms[@]}";
  foo bar foo
```

**test/read_unique.t**

```
Sets unique arguments into a variable

  $ read_unique X x y y z y z
  > echo "${X[@]}"
  x y z
```

WDYT?

Reply:
True, it makes sense to disentangle the functions.

Although I might add that `test/options_sanity/options_sanity.t` already tests option parsing similar to your first suggestion, albeit without duplicate packages/cache-directories.

In that case should we just replace `test/options_sanity/array_uniqueness.t` with your second suggestion?

Reply:
> we just replace test/options_sanity/array_uniqueness.t with your second suggestion?

Sure, SGTM.